local-pv-provisioner/controllers/persistent_volume.go

```diff
@@ -135,9 +129,7 @@ func (w *persistentVolumeWatcher) fireEvent(ctx context.Context) error {
 			continue
 		}
 		w.ch <- event.GenericEvent{
-			Meta: &metav1.ObjectMeta{
-				Name: pv.Name,
-			},
+			Object: &pv,
```

Comment:
`pv` should not be passed as a pointer.

You can see the strange behavior in the following code.

```go
package main

import (
	"fmt"
	"time"
)

type GenericEvent struct {
	Object Object
}

type Object interface {
	GetName() string
	SetName(name string)
}

type hoge struct {
	name string
}

func(h *hoge) GetName() string{
	return h.name
}

func(h *hoge) SetName(name string) {
	h.name = name
}


func main(){
	ch := make(chan GenericEvent)

	data := []hoge{
		{
			name: "a",
		},
		{
			name: "b",
		},
		{
			name: "c",
		},
	}

	go func() {
		for _, d := range data {
			ch <- GenericEvent{
				Object:  &d,
			}
		}
	}()

	go func() {
		for {
			got := <- ch
			fmt.Println(got.Object.GetName())
		}
	}()

	time.Sleep(5*time.Second)
}
```